contracts/bridge/BridgeTransferERC721.sol

```diff
@@ -76,28 +76,49 @@ contract BridgeTransferERC721 is BridgeTokens, IERC721BridgeReceiver, BridgeTran
         address _from,
         address _to,
         uint256 _tokenId,
-        bytes memory _extraData
+        bytes memory _extraData,
+        uint ver
     )
         internal
         onlyRegisteredToken(_tokenAddress)
         onlyUnlockedToken(_tokenAddress)
     {
         require(isRunning, "stopped bridge");
+        require(ver == 1 || ver == 2, "Unknown version");
+
+        if (ver == 1) {
+            emit RequestValueTransfer(
+                TokenType.ERC721,
+                _from,
+                _to,
+                _tokenAddress,
+                _tokenId,
+                requestNonce,
+                0,
+                _extraData
+            );
+        } else {
+            (bool success, bytes memory uri) = _tokenAddress.call(abi.encodePacked(ERC721Metadata(_tokenAddress).tokenURI.selector, abi.encode(_tokenId)));
+            if (success == false) {
+                uri = "";
+            }
 
+            emit RequestValueTransferEncoded(
```

Comment:
You should move this after the burn. even if the burn fails, the event anyway occurs if it is here.
You can just declare a variable to store `abi.encode(string(uri))` here and then pass the variable to the event.

Reply:
The `burn()` function reverts if the function fails. Could you elaborate If your concern is not this?

Reply:
hmm.. yeah you're right. I thought that it can be open to reentrance attack, but it seems not.